File: android/lab6/untitled folder/task42/app/src/main/java/com/example/task42/MainActivity.java
package com.example.task42;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import android.content.ContentValues;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private DatabaseHelper dbHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        dbHelper = new DatabaseHelper(this);

        Button button1 = findViewById(R.id.button1);
        button1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                displayData();
            }
        });

        Button button2 = findViewById(R.id.button2);
        button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                insertData();
            }
        });

        Button button3 = findViewById(R.id.button3);
        button3.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                updateData();
            }
        });
    }

    private void displayData() {
        SQLiteDatabase db = dbHelper.getReadableDatabase();

        String[] projection = {
                "ID",
                "Brand",
                "SerialNumber",
                "Year"
        };

        Cursor cursor = db.query(
                "Car",
                projection,
                null,
                null,
                null,
                null,
                null
        );

        StringBuilder data = new StringBuilder();
        while (cursor.moveToNext()) {
            int id = cursor.getInt(cursor.getColumnIndexOrThrow("ID"));
            String brand = cursor.getString(cursor.getColumnIndexOrThrow("Brand"));
            String serialNumber = cursor.getString(cursor.getColumnIndexOrThrow("SerialNumber"));
            int year = cursor.getInt(cursor.getColumnIndexOrThrow("Year"));

            data.append("ID: ").append(id)
                    .append(", Brand: ").append(brand)
                    .append(", Serial Number: ").append(serialNumber)
                    .append(", Year: ").append(year)
                    .append("\n");
        }

        cursor.close();
        db.close();

        Toast.makeText(this, data.toString(), Toast.LENGTH_LONG).show();
    }

    private void insertData() {


        SQLiteDatabase db = dbHelper.getWritableDatabase();

        ContentValues values = new ContentValues();
        values.put("Brand", "Toyota");
        values.put("SerialNumber", "123456789");
        values.put("Year", 2022);

        long newRowId = db.insert("Car", null, values);

        if (newRowId != -1) {
            Toast.makeText(this, "Запись добавлена", Toast.LENGTH_SHORT).show();
        } else {
            Toast.makeText(this, "Ошибка при добавлении записи", Toast.LENGTH_SHORT).show();
        }

        db.close();
    }

    private void updateData() {
        SQLiteDatabase db = dbHelper.getWritableDatabase();

        ContentValues values = new ContentValues();
        values.put("Brand", "Honda");

        String selection = "ID = ?";
        String[] selectionArgs = { "1" };

        int count = db.update(
                "Car",
                values,
                selection,
                selectionArgs
        );

        if (count > 0) {
            Toast.makeText(this, "Запись изменена", Toast.LENGTH_SHORT).show();
        } else {
            Toast.makeText(this, "Ошибка при изменении записи", Toast.LENGTH_SHORT).show();
        }

        db.close();
    }

    @Override
    protected void onDestroy() {
        dbHelper.close();
        super.onDestroy();
    }
}

File: android/lab6/4.2/app/src/main/java/com/example/task2/MainActivity.kt
package com.example.task2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.content.ContentValues
import android.content.Intent
import android.widget.Button

// Constants
private const val TABLE_NAME = "Автомобиль"
private const val COLUMN_ID = "ID"
private const val COLUMN_MARK = "Марка"
private const val COLUMN_NUMBER = "Серийный_номер"
private const val COLUMN_YEAR = "Год_выпуска"

class MainActivity : AppCompatActivity() {

    private lateinit var dbHelper: DatabaseHelper
    private lateinit var buttonShowInfo: Button
    private lateinit var buttonAddRecord: Button
    private lateinit var buttonUpdateRecord: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        dbHelper = DatabaseHelper(this)
        createTableIfNotExists()

        buttonShowInfo = findViewById(R.id.buttonShowInfo)
        buttonAddRecord = findViewById(R.id.buttonAddRecord)
        buttonUpdateRecord = findViewById(R.id.buttonUpdateRecord)

        buttonShowInfo.setOnClickListener {
            val intent = Intent(this, InfoActivity::class.java)
            startActivity(intent)
        }

        buttonAddRecord.setOnClickListener {
            val db = dbHelper.writableDatabase
            val values = ContentValues().apply {
                put(COLUMN_MARK, "Toyota")
                put(COLUMN_NUMBER, "123456")
                put(COLUMN_YEAR, 2018)
            }
            db.insert(TABLE_NAME, null, values)
            db.close()
        }

        buttonUpdateRecord.setOnClickListener {
            val db = dbHelper.writableDatabase
            val values = ContentValues().apply {
                put(COLUMN_NUMBER, "789012")
                put(COLUMN_MARK, "BMW")
                put(COLUMN_YEAR, "2020")
            }
            db.update(TABLE_NAME, values, "$COLUMN_ID = ?", arrayOf("1"))
            db.close()
        }
    }

    private fun createTableIfNotExists() {
        val db = dbHelper.writableDatabase
        db.execSQL("CREATE TABLE IF NOT EXISTS $TABLE_NAME (" +
                "$COLUMN_ID INTEGER PRIMARY KEY AUTOINCREMENT," +
                "$COLUMN_MARK TEXT," +
                "$COLUMN_NUMBER TEXT," +
                "$COLUMN_YEAR INTEGER" +
                ")")

        db.use {
            val count = it.rawQuery("SELECT COUNT(*) FROM $TABLE_NAME", null).use { cursor ->
                cursor.moveToFirst()
                cursor.getInt(0)
            }

            if (count == 0) {
                // Удалить все записи, если существуют
                db.delete(TABLE_NAME, null, null)

                // Добавить 5 записей
                val carValues = arrayOf(
                    arrayOf("Mercedes", "345678", 2019),
                    arrayOf("Honda", "567890", 2017),
                    arrayOf("Ford", "901234", 2021),
                    arrayOf("Volkswagen", "890123", 2015),
                    arrayOf("Nissan", "012345", 2022)
                )

                for (car in carValues) {
                    val carData = ContentValues().apply {
                        put(COLUMN_MARK, car[0].toString())
                        put(COLUMN_NUMBER, car[1].toString())
                        put(COLUMN_YEAR, car[2].toString())
                    }
                    db.insert(TABLE_NAME, null, carData)
                }
            }
        }
    }

    override fun onDestroy() {
        dbHelper.close()
        super.onDestroy()
    }
}
